consensus/src/quorum_store/batch_aggregator.rs

```diff
@@ -117,6 +117,9 @@ impl BatchAggregator {
 
     fn is_outdated_fragment(&self, batch_id: BatchId, fragment_id: usize) -> bool {
         if let Some(self_batch_id) = self.batch_id {
+            if self_batch_id.nonce != batch_id.nonce {
+                return false;
+            }
```

Comment:
Should this case be missed fragment or an outdated fragment?
Should not we store all previous nonces in this epoch and compare them against all of them?

Reply:
Actually, why do we even check it here to return false?